x-pack/plugins/cases/server/routes/api/utils.test.ts

```diff
@@ -54,4 +54,55 @@ describe('Utils', () => {
       expect(res.headers).toEqual({});
     });
   });
+  describe('getCaseUrl', () => {
+    const caseId = '12345';
+    const baseUrl = 'https://kibana.comma/app/security';
+    const baseObsUrl = 'https://kibana.comma/app/observability';
+    const mockUrlState =
+      '?sourcerer=(default:!(%27auditbeat-*%27))&timerange=(global:(linkTo:!(timeline),timerange:(from:%272021-06-15T06:00:00.000Z%27,fromStr:now%2Fd,kind:relative,to:%272021-06-16T05:59:59.999Z%27,toStr:now%2Fd)),timeline:(linkTo:!(global),timerange:(from:%272021-06-15T06:00:00.000Z%27,fromStr:now%2Fd,kind:relative,to:%272021-06-16T05:59:59.999Z%27,toStr:now%2Fd)))';
+    const createUrl = `${baseUrl}/cases/create`;
+    const detectionsUrl = `${baseUrl}/detections`;
+    const caseUrl = `${baseUrl}/cases/12345`;
+    const createObsUrl = `${baseObsUrl}/cases/create`;
+    const overviewObsUrl = `${baseObsUrl}/overview`;
+    const caseObsUrl = `${baseObsUrl}/cases/12345`;
+    describe('security urls', () => {
+      it('gets case details url from case details link', () => {
+        const result = getCaseUrl(`${caseUrl}${mockUrlState}`, caseId);
+        expect(result).toEqual(caseUrl);
+      });
+      it('gets case details url from create link', () => {
+        const result = getCaseUrl(`${createUrl}${mockUrlState}`, caseId);
+        expect(result).toEqual(caseUrl);
+      });
+      it('gets case details url from detections link', () => {
+        const result = getCaseUrl(`${detectionsUrl}${mockUrlState}`, caseId);
+        expect(result).toEqual(caseUrl);
+      });
+    });
+    describe('observability urls', () => {
+      it('gets case details url from case details link', () => {
+        const result = getCaseUrl(`${caseObsUrl}${mockUrlState}`, caseId);
+        expect(result).toEqual(caseObsUrl);
+      });
+      it('gets case details url from create link', () => {
+        const result = getCaseUrl(`${createObsUrl}${mockUrlState}`, caseId);
+        expect(result).toEqual(caseObsUrl);
+      });
+      it('gets case details url from detections link', () => {
+        const result = getCaseUrl(`${overviewObsUrl}${mockUrlState}`, caseId);
+        expect(result).toEqual(caseObsUrl);
+      });
+    });
+    describe('bad urls', () => {
+      it('returns null from random link', () => {
+        const result = getCaseUrl(`https://not-an-offical-kibana.com`, caseId);
+        expect(result).toEqual(null);
+      });
+      it('non security/obs app', () => {
+        const result = getCaseUrl(`'https://kibana.comma/app/uptime'`, caseId);
```

Comment:
if these apps start creating cases, we'll need to update the `getCaseUrl` function to handle intelligently